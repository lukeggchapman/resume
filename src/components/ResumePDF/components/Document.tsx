import React from 'react'
import {
  Page,
  Text,
  Link as PDFLink,
  View,
  Document,
  StyleSheet,
} from '@react-pdf/renderer'

import typography, { FAIcon } from './typography'
import Section from './common/Section'
import Experience from './Experience'
import Education from './Education'
import Skills from './Skills'
import ResumeData from '../ResumeData'

export interface DocumentProps {
  data: ResumeData
}

const stylesObj = {
  page: {
    flexDirection: 'column',
    backgroundColor: 'white',
    fontFamily: 'Crimson Text',
    fontSize: 12,
    color: '#333',
    padding: '20 30',
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 10,
  },
  info: {
    alignItems: 'right',
    fontFamily: 'PT Sans',
  },
  body: {
    flexDirection: 'row',
    flexGrow: 1,
  },
  content: {
    flex: 6,
  },
  sidePanel: {
    flex: 1,
    marginLeft: 10,
  },
  footer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
}
const styles = StyleSheet.create(stylesObj)

const ResumeDocument = ({ data }: DocumentProps) => {
  return (
    <Document title="Luke Chapman Résumé">
      <Page size="A4" style={styles.page}>
        <View style={styles.header}>
          <Text style={typography.h1}>Luke Chapman</Text>
          <View style={styles.info}>
            <Text>
              <FAIcon icon="envelope" fas />
              {` `}
              <PDFLink src="mailto:lukeggchapman@gmail.com">
                LukeGGChapman@gmail.com
              </PDFLink>
            </Text>
            <Text>
              <FAIcon icon="phone" fas />
              {` `}
              <PDFLink src="tel:+61401516361">+61 401 516 361</PDFLink>
            </Text>
          </View>
        </View>
        <View style={styles.body}>
          <View style={styles.content}>
            <Section title="Experience">
              <Experience data={data.experience} />
            </Section>
            <Section title="Education">
              <Education data={data.education} />
            </Section>
          </View>
          <View style={styles.sidePanel}>
            <Section title="Skills">
              <Skills data={data.skills} />
            </Section>
          </View>
        </View>
        <View style={[typography.h4, styles.footer]}>
          <Text>
            <FAIcon icon="github" fab />
            {` `}
            lukeggchapman
          </Text>
          <Text>
            <FAIcon icon="linkedin" fab />
            {` `}
            lukeggchapman
          </Text>
          <Text>
            <FAIcon icon="react" fab />
            {` `}
            Created with React.
            {' Sourcecode '}
            <PDFLink src="https://bit.ly/2VgE4pH">
              https://bit.ly/2VgE4pH
            </PDFLink>
            {` `}
            <PDFLink src="" />
          </Text>
        </View>
      </Page>
    </Document>
  )
}

export default ResumeDocument
