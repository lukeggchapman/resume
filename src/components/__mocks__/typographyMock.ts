import idObj from 'identity-obj-proxy'

jest.doMock('../typography', () => ({
  __esModule: true,
  default: idObj,
  FAIcon: 'FAIcon',
}))
