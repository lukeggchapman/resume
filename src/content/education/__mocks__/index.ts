const mockEducationData = [
  {
    frontmatter: {
      school: 'schoolTest1',
      endDate: '2004-05-18',
      degree: 'degreeTest1',
      startDate: '2001-01-11',
    },
    logo: 'logoPublcUrlTest1',
    htmlAst: 'htmlAstTest1' as any,
  },
]

export default mockEducationData
