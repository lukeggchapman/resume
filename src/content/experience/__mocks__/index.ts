const mockExperienceData = [
  {
    frontmatter: {
      company: 'companyTest1',
      endDate: 'endDateTest1',
      role: 'roleTest1',
      startDate: 'startDateTest1',
    },
    logo: 'logoPublcUrlTest1',
    htmlAst: 'htmlAstTest1' as any,
  },
  {
    frontmatter: {
      company: 'companyTest2',
      endDate: 'endDateTest2',
      role: 'roleTest2',
      startDate: 'startDateTest2',
    },
    logo: 'logoPublcUrlTest2',
    htmlAst: 'htmlAstTest2' as any,
  },
]

export default mockExperienceData
