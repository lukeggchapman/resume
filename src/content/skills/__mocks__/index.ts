export default {
  beginner: ['skillTest1', 'skillTest2'],
  intermediate: ['skillTest3', 'skillTest4'],
  advanced: ['skillTest5', 'skillTest6'],
}
