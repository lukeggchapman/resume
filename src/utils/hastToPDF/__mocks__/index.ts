export default jest.fn((htmlAst) => htmlAst)
