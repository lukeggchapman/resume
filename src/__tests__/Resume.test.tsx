import React from 'react'
import { shallow } from 'enzyme'
import '../components/__mocks__/typographyMock'
import Document from '../Resume'

jest.mock('sections/Experience', () => 'Experience')
jest.mock('sections/Education', () => 'Education')
jest.mock('sections/Skills', () => 'Skills')

describe('Document', () => {
  const render = () => shallow(<Document />)

  it('renders correctly', () => {
    expect(render()).toMatchSnapshot()
  })
})
