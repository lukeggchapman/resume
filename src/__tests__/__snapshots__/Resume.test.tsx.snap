// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document renders correctly 1`] = `
<div>
  <div
    data-pdf="Document"
    title="Luke Chapman Resume"
  >
    <div
      data-pdf="Page"
      data-style="[
  "body",
  {
    "flexDirection": "column",
    "backgroundColor": "white",
    "padding": "20 30"
  }
]"
    >
      <div
        data-pdf="View"
        data-style="{
  "flexDirection": "row",
  "justifyContent": "space-between",
  "alignItems": "center",
  "marginBottom": 10
}"
      >
        <div
          data-pdf="Text"
          data-style=""h1""
        >
          Luke Chapman
        </div>
        <div
          data-pdf="View"
          data-style="{
  "alignItems": "flex-end",
  "fontFamily": "Noto Sans",
  "fontSize": 10
}"
        >
          <div
            data-pdf="Text"
          >
            <div
              data-pdf="Text"
              data-style="[
  {
    "fontFamily": "Font Awesome 5 Free"
  },
  {
    "fontWeight": 900
  }
]"
            >
              
            </div>
             
            <div
              data-pdf="Link"
              src="mailto:lukeggchapman@gmail.com"
            >
              lukeggchapman@gmail.com
            </div>
          </div>
          <div
            data-pdf="Text"
          >
            <div
              data-pdf="Text"
              data-style="[
  {
    "fontFamily": "Font Awesome 5 Free"
  },
  {
    "fontWeight": 900
  }
]"
            >
              
            </div>
             
            <div
              data-pdf="Link"
              src="tel:+61401516361"
            >
              0401516361
            </div>
          </div>
        </div>
      </div>
      <div
        data-pdf="View"
        data-style="{
  "flexDirection": "row",
  "flexGrow": 1
}"
      >
        <div
          data-pdf="View"
          data-style="{
  "flex": 6
}"
        >
          <div
            data-pdf="View"
          >
            <div
              data-pdf="Text"
              data-style=""h2""
            >
              Experience
            </div>
            <div
              data-pdf="View"
              data-style="[
  {
    "flexDirection": "column",
    "fontSize": 8,
    "padding": "10 6"
  },
  {}
]"
            >
              <div
                data-pdf="View"
                data-style="{
  "flexDirection": "row",
  "marginBottom": 4
}"
              >
                <div
                  data-pdf="View"
                  data-style="{
  "alignItems": "center",
  "width": 45
}"
                >
                  <div
                    data-pdf="Image"
                    data-style="{
  "height": 35,
  "paddingRight": 10
}"
                    src="logoPublcUrlTest1"
                  />
                </div>
                <div
                  data-pdf="View"
                  data-style="{
  "flexGrow": 1,
  "justifyContent": "center"
}"
                >
                  <div
                    data-pdf="Text"
                    data-style=""h3""
                  >
                    companyTest1
                  </div>
                  <div
                    data-pdf="View"
                    data-style="[
  "h4",
  {
    "flexDirection": "row",
    "justifyContent": "space-between"
  }
]"
                  >
                    <div
                      data-pdf="Text"
                    >
                      roleTest1
                    </div>
                    <div
                      data-pdf="View"
                    >
                      <div
                        data-pdf="Text"
                      >
                        Jun, 2021 – Apr, 2020
                         · 
                        -2 yr -1 mo
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                data-pdf="View"
                data-style="{
  "flexDirection": "column",
  "width": "100%",
  "padding": 10
}"
              >
                <div
                  data-pdf="View"
                  data-style="{
  "marginTop": 5
}"
                  data-testid="content"
                >
                  htmlAstTest1
                </div>
              </div>
              <div
                data-pdf="Image"
                data-style="{
  "width": 400,
  "alignSelf": "center",
  "marginTop": 6
}"
                src="test-file-stub"
              />
            </div>
            <div
              data-pdf="View"
              data-style="[
  {
    "flexDirection": "column",
    "fontSize": 8,
    "padding": "10 6"
  },
  {}
]"
            >
              <div
                data-pdf="View"
                data-style="{
  "flexDirection": "row",
  "marginBottom": 4
}"
              >
                <div
                  data-pdf="View"
                  data-style="{
  "alignItems": "center",
  "width": 45
}"
                >
                  <div
                    data-pdf="Image"
                    data-style="{
  "height": 35,
  "paddingRight": 10
}"
                    src="logoPublcUrlTest2"
                  />
                </div>
                <div
                  data-pdf="View"
                  data-style="{
  "flexGrow": 1,
  "justifyContent": "center"
}"
                >
                  <div
                    data-pdf="Text"
                    data-style=""h3""
                  >
                    companyTest2
                  </div>
                  <div
                    data-pdf="View"
                    data-style="[
  "h4",
  {
    "flexDirection": "row",
    "justifyContent": "space-between"
  }
]"
                  >
                    <div
                      data-pdf="Text"
                    >
                      roleTest2
                    </div>
                    <div
                      data-pdf="View"
                    >
                      <div
                        data-pdf="Text"
                      >
                        Jul, 2021 – Aug, 2023
                         · 
                        2 yrs 1 mo
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                data-pdf="View"
                data-style="{
  "flexDirection": "column",
  "width": "100%",
  "padding": 10
}"
              >
                <div
                  data-pdf="View"
                  data-style="{
  "marginTop": 5
}"
                  data-testid="content"
                >
                  htmlAstTest2
                </div>
              </div>
              <div
                data-pdf="Image"
                data-style="{
  "width": 400,
  "alignSelf": "center",
  "marginTop": 6
}"
                src="test-file-stub"
              />
            </div>
          </div>
          <div
            data-pdf="View"
          >
            <div
              data-pdf="Text"
              data-style=""h2""
            >
              Education
            </div>
            <div
              data-pdf="View"
              data-style="[
  {
    "flexDirection": "column",
    "fontSize": 8,
    "padding": "10 6"
  },
  {}
]"
            >
              <div
                data-pdf="View"
                data-style="{
  "flexDirection": "row",
  "marginBottom": 4
}"
              >
                <div
                  data-pdf="View"
                  data-style="{
  "alignItems": "center",
  "width": 45
}"
                >
                  <div
                    data-pdf="Image"
                    data-style="{
  "height": 35,
  "paddingRight": 10
}"
                    src="logoPublcUrlTest1"
                  />
                </div>
                <div
                  data-pdf="View"
                  data-style="{
  "flexGrow": 1,
  "justifyContent": "center"
}"
                >
                  <div
                    data-pdf="Text"
                    data-style=""h3""
                  >
                    schoolTest1
                  </div>
                  <div
                    data-pdf="View"
                    data-style="[
  "h4",
  {
    "flexDirection": "row",
    "justifyContent": "space-between"
  }
]"
                  >
                    <div
                      data-pdf="Text"
                    >
                      degreeTest1
                    </div>
                    <div
                      data-pdf="View"
                    >
                      <div
                        data-pdf="Text"
                      >
                        Jan, 2001 – May, 2004
                         · 
                        3 yrs 4 mos
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                data-pdf="View"
                data-style="{
  "flexDirection": "column",
  "width": "100%",
  "padding": 10
}"
              >
                <div
                  data-pdf="View"
                  data-style="{
  "marginTop": 5
}"
                  data-testid="content"
                >
                  htmlAstTest1
                </div>
              </div>
              <div
                data-pdf="Image"
                data-style="{
  "width": 400,
  "alignSelf": "center",
  "marginTop": 6
}"
                src="test-file-stub"
              />
            </div>
          </div>
        </div>
        <div
          data-pdf="View"
          data-style="{
  "flex": 1,
  "marginLeft": 20
}"
        >
          <div
            data-pdf="View"
          >
            <div
              data-pdf="Text"
              data-style=""h2""
            >
              Skills
            </div>
            <div
              data-pdf="View"
              data-style="{
  "marginBottom": 10
}"
            >
              <div
                data-pdf="Text"
                data-style="[
  "h3",
  {
    "textTransform": "capitalize",
    "marginBottom": 5
  }
]"
              >
                beginner
              </div>
              <div
                data-pdf="Text"
                data-style="{
  "color": "#333",
  "fontFamily": "PT Sans",
  "fontSize": 8
}"
              >
                skillTest1
              </div>
              <div
                data-pdf="Text"
                data-style="{
  "color": "#333",
  "fontFamily": "PT Sans",
  "fontSize": 8
}"
              >
                skillTest2
              </div>
            </div>
            <div
              data-pdf="View"
              data-style="{
  "marginBottom": 10
}"
            >
              <div
                data-pdf="Text"
                data-style="[
  "h3",
  {
    "textTransform": "capitalize",
    "marginBottom": 5
  }
]"
              >
                intermediate
              </div>
              <div
                data-pdf="Text"
                data-style="{
  "color": "#333",
  "fontFamily": "PT Sans",
  "fontSize": 8
}"
              >
                skillTest3
              </div>
              <div
                data-pdf="Text"
                data-style="{
  "color": "#333",
  "fontFamily": "PT Sans",
  "fontSize": 8
}"
              >
                skillTest4
              </div>
            </div>
            <div
              data-pdf="View"
              data-style="{
  "marginBottom": 10
}"
            >
              <div
                data-pdf="Text"
                data-style="[
  "h3",
  {
    "textTransform": "capitalize",
    "marginBottom": 5
  }
]"
              >
                advanced
              </div>
              <div
                data-pdf="Text"
                data-style="{
  "color": "#333",
  "fontFamily": "PT Sans",
  "fontSize": 8
}"
              >
                skillTest5
              </div>
              <div
                data-pdf="Text"
                data-style="{
  "color": "#333",
  "fontFamily": "PT Sans",
  "fontSize": 8
}"
              >
                skillTest6
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        data-pdf="View"
        data-style="[
  "h4",
  {
    "flexDirection": "row",
    "justifyContent": "space-between"
  }
]"
      >
        <div
          data-pdf="Text"
        >
          <div
            data-pdf="Text"
            data-style="[
  {
    "fontFamily": "Font Awesome 5 Free"
  },
  {
    "fontWeight": 700
  }
]"
          >
            
          </div>
           
          <div
            data-pdf="Link"
            src="https://github.com/lukeggchapman"
          >
            lukeggchapman
          </div>
        </div>
        <div
          data-pdf="Text"
        >
          <div
            data-pdf="Text"
            data-style="[
  {
    "fontFamily": "Font Awesome 5 Free"
  },
  {
    "fontWeight": 700
  }
]"
          >
            
          </div>
           
          <div
            data-pdf="Link"
            src="https://www.linkedin.com/in/lukeggchapman"
          >
            lukeggchapman
          </div>
        </div>
        <div
          data-pdf="Text"
        >
          <div
            data-pdf="Text"
            data-style="[
  {
    "fontFamily": "Font Awesome 5 Free"
  },
  {
    "fontWeight": 700
  }
]"
          >
            
          </div>
           
          Created with React.
           Source code 
          <div
            data-pdf="Link"
            src="https://bit.ly/2VgE4pH"
          >
            https://bit.ly/2VgE4pH
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
